core/presentation/designsystem/src/main/java/com/nolawiworkineh/designsystem/components/TaskyToolBar.kt

```diff
@@ -0,0 +1,231 @@
+@file:OptIn(ExperimentalMaterial3Api::class)
+
+package com.nolawiworkineh.designsystem.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Delete
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material.icons.filled.Settings
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.material3.TopAppBarScrollBehavior
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.nolawiworkineh.designsystem.components.util.DropDownMenuItem
+
+
+@Composable
+fun TaskyTopAppBar(
+    title: String = "", // Default string title
+    showBackButton: Boolean = false,
+    isBlackToolBar: Boolean = true, // Determines whether toolbar is black
```

Comment:
Not necessary for this project, but in general it would be a more flexible approach to make this parameter a `toolbarColor: Color = Color.Black`. Still shows the default value, but also let's you pick a different color and does not hide what kind of color it would be if it's not black